### PR TITLE
En_Trt2

### PR DESCRIPTION
**Pull request by Kelebek1**
Before opening this PR, ensure the following:
- `./format.sh` was run to apply standard formatting.
- `make` successfully builds a matching ROM.
- No new compiler warnings were introduced during the build process.
    - Can be verified locally by running `tools/warnings_count/check_new_warnings.sh`
- New variables & functions should follow standard naming conventions.
- Comments and variables have correct spelling.
---

### CODE REVIEW

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 1]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+    } else if (gSaveContext.weekEventReg[16] & 0x10) {
+        this->unk_3A8 = 0x838;
+        return;
+    } else if (gSaveContext.weekEventReg[17] & 1) {
+        this->unk_3A8 = 0x84D;
+        return;
+    } else {
+        this->unk_3A8 = 0x849;
+    }
+}
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:157]: Are all these returns needed?

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 2]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+    this->actor.world.pos.y += D_80AD5904.y;
+    this->actor.world.pos.z += D_80AD5904.z;
+
+    ActorShape_Init(&this->actor.shape, 0.0f, func_800B3FC0, 20.0f);
+    func_80AD469C(this, globalCtx);
+    this->actor.colChkInfo.mass = MASS_IMMOVABLE;
+    this->actor.colChkInfo.cylRadius = 50;
+
+    this->unk_3C0 = 0;
+    this->unk_3D8 = 0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:738]:
```suggestion
    this->unk_3D8 = false;
```
This variable is boolean

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 3]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+    }
+}
+
+void func_80AD3CEC(EnTrt2* this, GlobalContext* globalCtx) {
+    u8 sp27 = func_80152498(&globalCtx->msgCtx);
+
+    func_80AD46F8(this);
+    if (this->unk_3D8 != 0) {
+        func_801518B0(globalCtx, this->unk_3A8, &this->actor);
+        this->unk_3D8 = 0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:346]:
```suggestion
    if (this->unk_3D8) {
        func_801518B0(globalCtx, this->unk_3A8, &this->actor);
        this->unk_3D8 = false;
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 4]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+                this->unk_3A8 = 0x84F;
+            }
+            func_80AD3DA4(this, globalCtx);
+        } else if ((player->transformation == PLAYER_FORM_HUMAN) ||
+                   (player->transformation == PLAYER_FORM_FIERCE_DEITY)) {
+            this->unk_3B2 = 16;
+            this->actor.velocity.y = 1.5f;
+        } else {
+            this->unk_3A8 = 0x84F;
+            this->unk_3D8 = 1;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:679]:
```suggestion
            this->unk_3D8 = true;
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 5]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+    player->stateFlags1 &= ~0x20;
+    Actor_MarkForDeath(&this->actor);
+}
+
+void func_80AD434C(EnTrt2* this, GlobalContext* globalCtx) {
+    this->actor.shape.rot.y += 0x1000;
+    Math_ApproachF(&this->actor.velocity.y, 1.5f, 0.2f, 0.1f);
+
+    if (this->actor.world.pos.y > 200.0f) {
+        if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame) != 0) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:492]:
```suggestion
        if (Animation_OnFrame(&this->skelAnime, this->skelAnime.endFrame)) {
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 6]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+        points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+        points = &points[arg1];
+        phi_f14 = points->x - arg2->x;
+        phi_f12 = points->z - arg2->z;
+    } else {
+        phi_f14 = 0.0f;
+        phi_f12 = 0.0f;
+    }
+    *arg3 = (phi_f14 * phi_f14) + (phi_f12 * phi_f12);
+    return Math_Acot2F(phi_f12, phi_f14) * 10430.378f;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:617]:
```suggestion
    return RADF_TO_BINANG(Math_Acot2F(phi_f12, phi_f14));
```
Hopefully there's enough digits of accuracy for this to match

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 7]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+    if (path != NULL) {
+        points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+        points = &points[arg1];
+        phi_f14 = points->x - arg2->x;
+        phi_f12 = points->z - arg2->z;
+    } else {
+        phi_f14 = 0.0f;
+        phi_f12 = 0.0f;
+    }
+    *arg3 = (phi_f14 * phi_f14) + (phi_f12 * phi_f12);
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:616]:
```suggestion
    *arg3 = SQ(phi_f14) +SQ (phi_f12);
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 8]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+        ret = false;
+    }
+    return ret;
+}
+
+s32 func_80AD4B4C(EnTrt2* this, GlobalContext* globalCtx) {
+    s32 sp24 = false;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (func_800B84D0(&this->actor, globalCtx) != 0) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:663]:
```suggestion
    if (func_800B84D0(&this->actor, globalCtx)) {
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 9]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+
+    if (((this->unk_3B2 == 4) || (this->unk_3B2 == 5)) && this->actor.isTargeted && !(this->actor.bgCheckFlags & 1) &&
+        ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_FIERCE_DEITY))) {
+        this->actor.speedXZ = 0.0f;
+        this->actor.velocity.y = 1.5f;
+        this->unk_3B2 = 16;
+        return true;
+    }
+
+    if (func_80AD4C4C(this) && this->actor.isTargeted && (sp1E < 0x4000) && (sp1E >= -0x3FFF)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:710]:
```suggestion
    if (func_80AD4C4C(this) && this->actor.isTargeted && (sp1E < 0x4000) && (sp1E > -0x4000)) {
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 10]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+    this->unk_3B8 = 0;
+    this->unk_3BC = func_80AD4608;
+
+    if (gSaveContext.weekEventReg[12] & 8) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    } else if (gSaveContext.weekEventReg[84] & 0x40) {
+        Actor_MarkForDeath(&this->actor);
+        return;
+    } else if ((globalCtx->sceneNum == SCENE_20SICHITAI) || (globalCtx->sceneNum == SCENE_20SICHITAI2)) {
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:750]:
```suggestion
    if (gSaveContext.weekEventReg[12] & 8) {
        Actor_MarkForDeath(&this->actor);
        return;
    } 
    
    if (gSaveContext.weekEventReg[84] & 0x40) {
        Actor_MarkForDeath(&this->actor);
        return;
    } 
    
    if ((globalCtx->sceneNum == SCENE_20SICHITAI) || (globalCtx->sceneNum == SCENE_20SICHITAI2)) {
``
If you're returning, then you don't need the `else if`, they can just be `if` with a new line

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 11]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+}
+
+void EnTrt2_PostLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3s* rot, Actor* thisx) {
+    EnTrt2* this = THIS;
+    Vec3f sp30 = { 0.0f, -30.0f, 0.0f };
+    s32 phi_v0 = 0;
+
+    if (this->unk_3B2 >= 8) {
+        phi_v0 = 1;
+    }
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:902]:
```suggestion
    s32 phi_v0 = false;

    if (this->unk_3B2 >= 8) {
        phi_v0 = true
    }
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 12]
```diff
@@ -76,111 +116,832 @@ static DamageTable D_80AD58D8 = {
...
+    MtxF sp28;
+
+    Matrix_MultiplyVector3fByState(&sp70, &sp7C);
+    Matrix_CopyCurrentState(&sp28);
+    func_8018219C(&sp28, &sp68, 0);
+
+    *arg2 = sp7C;
+
+    if (arg4 != 0) {
+        sp68.x += arg0;
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:872]:
```suggestion
    if (arg4) {
        sp68.x += arg0;
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.h [hunk 13]
```diff
@@ -6,11 +6,39 @@
...
+    /* 0x148 */ char unk148[8];
+    /* 0x150 */ SkelAnime skelAnime;
+    /* 0x194 */ ColliderCylinder collider;
+    /* 0x1E0 */ Path* path;
+    /* 0x1E4 */ s32 unk_1E4;
+    /* 0x1E8 */ char unk1E8[0x154];
+    /* 0x33C */ s16 unk_33C[27];
+    /* 0x372 */ s16 unk_372[27];
+    /* 0x3A8 */ u16 unk_3A8;
+    /* 0x3AA */ char unk3AA[4];
```
Comment (engineer124) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.h:26]:
```suggestion
    /* 0x148 */ UNK_TYPE1 unk148[8];
    /* 0x150 */ SkelAnime skelAnime;
    /* 0x194 */ ColliderCylinder collider;
    /* 0x1E0 */ Path* path;
    /* 0x1E4 */ s32 unk_1E4;
    /* 0x1E8 */ UNK_TYPE1 unk1E8[0x154];
    /* 0x33C */ s16 unk_33C[27];
    /* 0x372 */ s16 unk_372[27];
    /* 0x3A8 */ u16 unk_3A8;
    /* 0x3AA */ UNK_TYPE1 unk3AA[4];
```

**Review by engineer124 — COMMENTED** [review]
Nice work. Some comments

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 14]
```diff
@@ -76,111 +103,832 @@ static DamageTable D_80AD58D8 = {
...
+    f32 phi_f12;
+    s32 ret;
+    f32 phi_f14;
+    s32 arg = arg2;
+    f32 sp44;
+    f32 sp40;
+    f32 sp3C;
+    Vec3f sp30;
+
+    points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:561]: The cast isn't necessary here
```suggestion
    points = Lib_SegmentedToVirtual(path->points);
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 15]
```diff
@@ -76,111 +103,832 @@ static DamageTable D_80AD58D8 = {
...
+    return ret;
+}
+
+s16 func_80AD48F8(Path* path, s32 arg1, Vec3f* arg2, f32* arg3) {
+    Vec3s* points;
+    f32 phi_f14;
+    f32 phi_f12;
+
+    if (path != NULL) {
+        points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:591]:
```suggestion
        points = Lib_SegmentedToVirtual(path->points);
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 16]
```diff
@@ -76,111 +103,832 @@ static DamageTable D_80AD58D8 = {
...
+    return RADF_TO_BINANG(Math_Acot2F(phi_f12, phi_f14));
+}
+
+f32 func_80AD49B8(Path* path, s32 arg1, Vec3f* arg2, Vec3s* arg3) {
+    s32 pad;
+    Vec3f sp20;
+    Vec3s* temp_v1;
+
+    if (path != NULL) {
+        temp_v1 = (Vec3s*)Lib_SegmentedToVirtual(path->points);
```
Comment (AngheloAlf) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:609]:
```suggestion
        temp_v1 = Lib_SegmentedToVirtual(path->points);
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 17]
```diff
@@ -76,111 +103,832 @@ static DamageTable D_80AD58D8 = {
...
+
+    points = (Vec3s*)Lib_SegmentedToVirtual(path->points);
+    count = path->count;
+    ret = false;
+    Math_Vec3s_ToVec3f(&sp30, &points[arg]);
+
+    if (arg == 0) {
+        phi_f12 = points[1].x - points[0].x;
+        phi_f14 = points[1].z - points[0].z;
+    } else if ((u32)count == (u32)(arg + 1)) {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:569]:
```suggestion
    } else if (arg == (count - 1) {
```
Does this work instead? Everywhere else I've seen this function, it gets repeated a lot, this works.

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 18]
```diff
@@ -76,111 +103,832 @@ static DamageTable D_80AD58D8 = {
...
+        phi_f12 = points[count - 1].x - points[count - 2].x;
+        phi_f14 = points[count - 1].z - points[count - 2].z;
+    } else {
+        phi_f12 = points[arg + 1].x - points[arg - 1].x;
+        phi_f14 = points[arg + 1].z - points[arg - 1].z;
+    }
+
+    func_8017B7F8(&sp30, func_80086B30(phi_f12, phi_f14) * 10430.378f, &sp44, &sp40, &sp3C);
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:577]:
```suggestion
    func_8017B7F8(&sp30, RADF_TO_BINANG(func_80086B30(phi_f12, phi_f14)), &sp44, &sp40, &sp3C);
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 19]
```diff
@@ -76,111 +103,832 @@ static DamageTable D_80AD58D8 = {
...
+        Matrix_InsertTranslation(this->unk_3C8.x, this->unk_3C8.y, this->unk_3C8.z, MTXMODE_NEW);
+        Matrix_Scale(this->actor.scale.x, this->actor.scale.y, this->actor.scale.z, MTXMODE_APPLY);
+        Matrix_RotateY(this->unk_3C2.y, MTXMODE_APPLY);
+        Matrix_InsertXRotation_s(this->unk_3C2.x, MTXMODE_APPLY);
+        Matrix_InsertZRotation_s(this->unk_3C2.z, MTXMODE_APPLY);
+    }
+}
+
+void func_80AD56E8(Actor* thisx, GlobalContext* globalCtx) {
+    static UNK_PTR D_80AD5978[] = {
```
Comment (hensldm) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:915]:
```suggestion
    static TEXTURE_PTR D_80AD5978[] = {
```

**Review by hensldm — COMMENTED** [review]
Some cosmetics

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c (hunk 19 above)
Comment (AngheloAlf) [reply src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:915]: It's `TexturePtr`, but yeah, please change it

Comment (hensldm) [reply src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:915]: Oops, my bad. Fixed in the suggestion.

**Review by hensldm — APPROVED** [review]

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 20]
```diff
@@ -76,111 +103,833 @@ static DamageTable D_80AD58D8 = {
...
+    }
+    return false;
+}
+
+s32 func_80AD4CCC(EnTrt2* this, GlobalContext* globalCtx) {
+    s16 sp1E = this->actor.yawTowardsPlayer - this->actor.shape.rot.y;
+    Player* player = GET_PLAYER(globalCtx);
+
+    if (((this->unk_3B2 == 4) || (this->unk_3B2 == 5)) && this->actor.isTargeted && !(this->actor.bgCheckFlags & 1) &&
+        ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_FIERCE_DEITY))) {
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:686]:
```suggestion
    if ((this->unk_3B2 == 4 || this->unk_3B2 == 5) && this->actor.isTargeted && 
        !(this->actor.bgCheckFlags & BGCHECK_CHECK_WALL) && 
        ((player->transformation == PLAYER_FORM_HUMAN) || (player->transformation == PLAYER_FORM_FIERCE_DEITY))) {
```
formatter probably still wont like that

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 21]
```diff
@@ -76,111 +103,833 @@ static DamageTable D_80AD58D8 = {
...
+}
+
+void func_80AD3530(EnTrt2* this, GlobalContext* globalCtx) {
+    s16 phi_a1;
+    f32 sp30;
+
+    func_80AD46F8(this);
+    if (this->path != NULL) {
+        phi_a1 = func_80AD48F8(this->path, this->unk_1E4, &this->actor.world.pos, &sp30);
+        if (this->actor.bgCheckFlags & 8) {
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:149]:
```suggestion
        if (this->actor.bgCheckFlags & BGCHECK_CHECK_ONE_FACE) {
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 22]
```diff
@@ -76,111 +103,833 @@ static DamageTable D_80AD58D8 = {
...
+        if (func_80AD475C(this, this->path, this->unk_1E4)) {
+            if (this->unk_1E4 >= (this->path->count - 1)) {
+                ActorCutscene_Stop(this->unk_3DA);
+                this->unk_3D9 = 2;
+            } else {
+                this->unk_1E4++;
+            }
+        }
+
+        if (this->actor.bgCheckFlags & 8) {
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:254]:
```suggestion
        if (this->actor.bgCheckFlags & BGCHECK_CHECK_ONE_FACE) {
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 23]
```diff
@@ -76,111 +103,833 @@ static DamageTable D_80AD58D8 = {
...
+    if (this->actor.world.pos.y < 5.0f) {
+        func_80AD3380(&this->skelAnime, sAnimations, 9);
+        this->unk_3B2 = 9;
+    }
+}
+
+void func_80AD3C94(EnTrt2* this, GlobalContext* globalCtx) {
+    if (this->actor.world.pos.y < 5.0f) {
+        func_80AD4A78(this, globalCtx);
+        if (this->actor.bgCheckFlags & 1) {
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:317]:
```suggestion
        if (this->actor.bgCheckFlags & BGCHECK_CHECK_WALL) {
```

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c [hunk 24]
```diff
@@ -76,111 +103,833 @@ static DamageTable D_80AD58D8 = {
     /* Powder Keg     */ DMG_ENTRY(1, 0x0),
 };
 
-// static InitChainEntry sInitChain[] = {
-static InitChainEntry D_80AD5964[] = {
+void func_80AD3380(SkelAnime* skelAnime, ActorAnimationEntryS* animation, s32 arg2) {
+    f32 phi_f0;
+
+    animation += arg2;
```
Comment (isghj5) [inline src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:109]: `animation += arg2` wat

**Review by isghj5 — COMMENTED** [review]
Minor documentation requests

File: src/overlays/actors/ovl_En_Trt2/z_en_trt2.c (hunk 24 above)
Comment (Kelebek1) [reply src/overlays/actors/ovl_En_Trt2/z_en_trt2.c:109]: Yeah this same form is used in a bunch of actors I've done. I can't get it to match using [], or using a temp, or using `animation= &animation[]` or anything either.

**Review by isghj5 — APPROVED** [review]
Sorry man

**Review by AngheloAlf — APPROVED** [review]

**Review by jpburnett — APPROVED** [review]

**Review by Kenix3 — APPROVED** [review]

**Pull request merged**